Taken the following patch from upstream cups changes:

commit 8ae6eb11184dcbd9eaf3c6badd4fad59fcc3863a
Author: Zdenek Dohnal <zdohnal@redhat.com>
Date:   Mon Aug 15 17:38:12 2022 +0200

    cups/dest.c: Write data into /etc/cups/lpoptions if we're root
    
    Fixes #454 , the patch is created by Yair Yarom
    (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1008053).

diff --git a/cups/dest.c b/cups/dest.c
index f563ce226..a9273ff93 100644
--- a/cups/dest.c
+++ b/cups/dest.c
@@ -2080,7 +2080,11 @@ cupsSetDests2(http_t      *http,	/* I - Connection to server or @code CUPS_HTTP_
 
   snprintf(filename, sizeof(filename), "%s/lpoptions", cg->cups_serverroot);
 
-  if (cg->home)
+  if (cg->home
+#ifndef _WIN32
+      && getuid() != 0
+#endif /* !_WIN32 */
+      )
   {
    /*
     * Create ~/.cups subdirectory...
